### PR TITLE
[iOS] Add sliding button.

### PR DESCRIPTION
**Pull request by sobisdav**
# :pencil: Description
- This PR adds an example of a sliding button that triggers action when used.

# :bulb: What’s new?
- Sliding button.

# :no_mouth: What’s missing?
- Nothing.

# :books: References
- Nothing.

### CODE REVIEW

File: ios/PresentationLayer/UIToolkit/Sources/UIToolkit/BaseViews/SwiftUI/Button/SlidingButton/SlidingButton.swift [hunk 1]
```diff
@@ -0,0 +1,67 @@
+//
+//  Created by David Sobíšek on 23.11.2023
+//  Copyright © 2023 Matee. All rights reserved.
+//
+
+import SwiftUI
+
+struct SlidingButton: View {
```
Comment (TereOnGit) [inline ios/PresentationLayer/UIToolkit/Sources/UIToolkit/BaseViews/SwiftUI/Button/SlidingButton/SlidingButton.swift]: public

File: ios/PresentationLayer/UIToolkit/Sources/UIToolkit/BaseViews/SwiftUI/Button/SlidingButton/SlidingButton.swift [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    
+    private let title: String
+    private let buttonIcon: Image
+    private let color: Color
+    private let isLoading: Bool
+    private let maxWidth: CGFloat
+    private let action: () -> Void
+    
+    init(
```
Comment (TereOnGit) [inline ios/PresentationLayer/UIToolkit/Sources/UIToolkit/BaseViews/SwiftUI/Button/SlidingButton/SlidingButton.swift]: public

File: ios/PresentationLayer/UIToolkit/Sources/UIToolkit/BaseViews/SwiftUI/Button/SlidingButton/SlidingButton.swift [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    ) {
+        self.title = title
+        self.buttonIcon = buttonIcon
+        self.color = color
+        self.isLoading = isLoading
+        self.maxWidth = maxWidth
+        self.action = action
+    }
+    
+    var body: some View {
```
Comment (TereOnGit) [inline ios/PresentationLayer/UIToolkit/Sources/UIToolkit/BaseViews/SwiftUI/Button/SlidingButton/SlidingButton.swift]: public

File: ios/PresentationLayer/UIToolkit/Sources/UIToolkit/BaseViews/SwiftUI/Button/SlidingButton/SlidingButton.swift [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        ZStack(alignment: .leading) {
+            RoundedRectangle(cornerRadius: 25)
+                .fill(color)
+                .frame(height: 50)
+            
+            Text(title)
```
Comment (TereOnGit) [inline ios/PresentationLayer/UIToolkit/Sources/UIToolkit/BaseViews/SwiftUI/Button/SlidingButton/SlidingButton.swift:39]: I would prefer text aligned on the center of slider :D

**Review by TereOnGit — COMMENTED** [review]
How do you feel about making AppThemes Dimens for CGFloats? Or at least making them as constants? @DavidKadlcek would approve 😄

**Comment by pchmelar** [comment]
👍 
An example of usage in Recipes would be nice 🙃

File: ios/PresentationLayer/UIToolkit/Sources/UIToolkit/BaseViews/SwiftUI/Button/SlidingButton/SlidingButton.swift (hunk 4 above)
Comment (sobisdav) [reply ios/PresentationLayer/UIToolkit/Sources/UIToolkit/BaseViews/SwiftUI/Button/SlidingButton/SlidingButton.swift:39]: In this example I prefer it where it is rn, you can put it in the middle when you use it elsewhere😀

**Review by pchmelar — APPROVED** [review]

**Pull request merged**